plugins/traction_innkeeper/traction_innkeeper/v1_0/tenant/routes.py

```diff
@@ -78,6 +80,33 @@ class TenantLedgerIdConfigSchema(OpenAPISchema):
         required=True,
     )
 
+@web.middleware
+async def setup_tenant_context(request: web.Request, handler):
+    """Middle ware to extract tenant_id and provide it to log formatter
+
+    This middleware is appended to the app middlewares and therefore runs
+    last. At this point the wallet_id has been extracted from a previous
+    middleware function and is used to query the tenant record.
+
+    Note: this function is very similar to tenant_self. Can we combine this logic?
+    """
+    context: AdminRequestContext = request["context"]
+    wallet_id, tenant_id = context.profile.settings.get("wallet.id"), None
+
+    if wallet_id:
+        mgr = context.inject(TenantManager)
+        profile = mgr.profile
+
+        async with profile.session() as session:
+            # Tenant records must always be fetched by their wallet id.
+            rec = await TenantRecord.query_by_wallet_id(session, wallet_id)
```

Comment:
Not sure if this gets cached or not? Because it's middleware it would be ideal if it was. It's just this traction plugin though so not a big deal.

Reply:
I think this is basically the same as my comment. Will do a performance assessment in `dev` and see if we need to address this at all.